src/app/auth/services/auth.service.ts

```diff
@@ -1,30 +1,82 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { Observable, BehaviorSubject } from "rxjs";
+import { SessionStorageService } from "./session-storage.service";
+import { User } from "@app/shared/models/user.model";
+import { tap } from "rxjs";
+
+import { Router } from "@angular/router";
+import { UserStoreService } from "@app/user/services/user-store.service";
+
+const TOKEN = "SESSION_TOKEN";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class AuthService {
-    login(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  private isAuthorized$$: BehaviorSubject<boolean>;
+  public isAuthorized$: Observable<boolean>;
+
+  constructor(
+    private http: HttpClient,
+    private sessionStorageService: SessionStorageService,
+    private router: Router,
+    private userStorageService: UserStoreService
+  ) {
+    this.isAuthorized$$ = new BehaviorSubject<boolean>(
+      this.sessionStorageService.hasToken()
+    );
+    this.isAuthorized$ = this.isAuthorized$$.asObservable();
+  }
+  login(user: User): Observable<any> {
+    // replace 'any' with the required interface
+    // Add your code here
+
+    return this.http.post<any>(this.getLoginUrl(), user).pipe(
+      tap((response) => {
+        if (response && response.result) {
+          this.sessionStorageService.setToken(response.result);
+          this.isAuthorized$$.next(true);
+
+          this.router.navigate(["/courses"]);
+        }
+      })
+    );
+  }
+
+  logout(): void {
+    // Add your code here
+    this.sessionStorageService.deleteToken();
+    this.isAuthorized$$.next(false);
+    this.router.navigate(["/login"]);
+  }
 
-    logout() {
-        // Add your code here
-    }
+  register(user: User): Observable<any> {
+    // replace 'any' with the required interface
+    // Add your code here
+    return this.http.post<any>(this.getRegisterUrl(), user);
+  }
 
-    register(user: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  get isAuthorised() {
+    // Add your code here. Get isAuthorized$$ value
+    return this.isAuthorized$$.getValue();
+  }
 
-    get isAuthorised() {
-        // Add your code here. Get isAuthorized$$ value
-    }
+  set isAuthorised(value: boolean) {
+    // Add your code here. Change isAuthorized$$ value
+    this.isAuthorized$$.next(value);
+  }
 
-    set isAuthorised(value: boolean) {
-        // Add your code here. Change isAuthorized$$ value
-    }
+  getLoginUrl() {
+    // Add your code here
+    return "http://localhost:4000/login";
```

Comment:
This is a classic example of managing environments in an Angular application. Suppose you are working on the development locally, but it will be deployed to some testing and a production environment later.
Angular has a designated place to manage it - under src/environments folder. You can add your URLs, env variables there individually. Also check the config in .angular file. Read more here: https://dev.to/remshams/replacing-configuration-files-in-angular-builds-based-on-build-targets-28jm